### ISSUE TITLE
SAM run/debug: support custom --build-dir instead of temporary dir

### ISSUE THREAD

**Issue by polovyivan** [aws/aws-toolkit-vscode#1988]
-   OS: Linux Ubuntu 18
-   VS Code version: 1.59.1
-   AWS Toolkit extension version: v1.29.0
-  SAM 1.29.0
-  The test source code is [here](https://github.com/polovyivan/aws-lambda-vs-code-debugging-and-local-test)

I would like to override the default sam build directory. 
For this, in the `launch.json`  file I set 

       "buildArguments": ["--build-dir", "${workspaceFolder}/.sam-build/"]

and then to point sam local invoke command to the template.yaml in a new build directory I set 

     "localArguments": ["--template","${workspaceFolder}/.sam-build/template.yaml"]

When I run/debug, the sam builds the source code and creates a template.yaml file and saves both to the specified directory (${workspaceFolder}/.sam-build/) but on the invoke stage it throws an error "template path does not exist: default template path>". Why it does not pick up the template file from the path I've specified in the localArguments?

![error_sam](https://user-images.githubusercontent.com/33532311/131164760-c1897370-1a97-496a-95da-d972fe955a4e.png)

Thanks!

**Comment by polovyivan** [comment]
Thanks for the fast response. I think this feature will be very useful. I prefer to keep everything in the project directory.
Just a little explanation of why I asked this question here, maybe someone will have the same problem. 
I could not make it work on my PC. 
The error I was having is this:
 
![error_sam_docker](https://user-images.githubusercontent.com/33532311/131202475-20fd6b02-cc9b-4cea-9570-530d7c238674.png)

The sam was throwing this error because it couldn't find in the docker container volume the file to run. 
I could not figure out why it is happening as everything was looking good. 
After checking literally everything I discovered that Docker on my PC was installed as an Ubuntu Snap. 
For some reason, Docker with this type of installation sees  `/tmp` as `tmp/snap.docker/tmp/`
When a Docker was running a container, instead of mounting volume with command displayed in the logs
`-v /tmp/aws-toolkit-vscode/vsctkXJybWE/output/testFunction:/var/task:ro`
it quietly mounted with the command below, which was creating a new directory on my PC, of course without any file in it.
`-v tmp/snap.docker/tmp/aws-toolkit-vscode/vsctkXJybWE/output/testFunction:/var/task:ro`
I reinstalled the Docker and everything works fine now.
